src/wallet/rpcwallet.cpp

```diff
@@ -5614,8 +5627,13 @@ UniValue z_mergetoaddress(const UniValue& params, bool fHelp)
 
     if (useAnyUTXO || taddrs.size() > 0) {
         // Get available utxos
-        vector<COutput> vecOutputs;
-        pwalletMain->AvailableCoins(vecOutputs, std::nullopt, true, NULL, false, false);
+        vector<COutput> vecOutputs = pwalletMain->AvailableCoins(
+            std::nullopt,
+            true,        // fOnlyConfirmed
+            nullptr,     // coinControl
+            false,       // fIncludeZeroValue
+            false,        // fIncludeCoinBase
+            ISMINE_SPENDABLE_ANY);
```

Comment:
Similarly:
```suggestion
            ISMINE_SPENDABLE_ANY,
            true);       // fOnlySpendable
```
and then removing the skipping logic below.